### PR TITLE
[BUGFIX beta] Make container always return the same value even if the registered value is falsy

### LINKED ISSUE #12796: Container doesn't return the same value if the registered value is falsy
According to [this issue](https://github.com/emberjs/ember.js/pull/3356), container can treat falsy values. However, on 2nd or later lookup, the container returns `undefined` even if the registered value is `false`, `null` or other falsy values. I believe this is not intentional.

Demo: [https://ember-twiddle.com/bc8eda6659978964e198](https://ember-twiddle.com/bc8eda6659978964e198?numColumns=3&openFiles=initializers.application.js%2Capplication.controller.js%2Capplication.template.hbs)

### PR DESCRIPTION
**Pull request by hibariya**
This fixes #12796.

### CODE REVIEW

**Comment by rwjblue** [comment]
Looks good to me. 

@dgeb / @stefanpenner - r?

File: packages/container/lib/registry.js [hunk 1]
```diff
@@ -849,12 +849,14 @@ function resolve(registry, normalizedName, options) {
     resolved = registry.resolver.resolve(normalizedName);
   }
 
-  resolved = resolved || registry.registrations[normalizedName];
+  if (resolved === undefined) {
```
Comment (stefanpenner) [inline packages/container/lib/registry.js]: isn't this expression the opposite of what we want?

Comment (hibariya) [inline packages/container/lib/registry.js]: I think at least this change is necessary for the following case:

```javascript
QUnit.test('The value returned from resolver is the same value as the original value even if the value is falsy', function() {
  let resolver = {
    resolve(fullName) {
      if (fullName === 'falsy:value') {
        return null;
      }
    }
  };
  let registry = new Registry({ resolver });

  deepEqual(registry.resolve('falsy:value'), null);
});
```

Comment (hibariya) [inline packages/container/lib/registry.js]: I added the above test.

**Comment by dgeb** [comment]
Looks good to me as well. Thanks @hibariya :+1:

**Comment by stefanpenner** [comment]
:+1:

**Pull request merged**